tests/locations/test_package.py

```diff
@@ -1370,6 +1370,91 @@ def test_move_uncompressed_aip(self):
         assert pkg.current_location == dst_location
 
 
+@pytest.fixture
+@pytest.mark.django_db
+def space(tmp_path):
+    space_dir = tmp_path / "space"
+    space_dir.mkdir()
+
+    staging_dir = tmp_path / "staging"
+    staging_dir.mkdir()
+
+    space = models.Space.objects.create(
+        access_protocol=models.Space.LOCAL_FILESYSTEM,
+        path=space_dir,
+        staging_path=staging_dir,
+    )
+    models.LocalFilesystem.objects.create(space=space)
+    return space
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def location(space):
+    aipstore = models.Location.objects.create(
+        space=space,
+        relative_path="fs-aips",
+        purpose="AS",
+    )
+    return aipstore
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def internal_location(space):
+    return models.Location.objects.create(
+        space=space, purpose=models.Location.STORAGE_SERVICE_INTERNAL, relative_path=""
+    )
+
+
+@pytest.fixture
+def bag_fixture(tmp_path):
+    tmp_dir = tmp_path / "dir"
+    tmp_dir.mkdir()
+    src = os.path.join(FIXTURES_DIR, "working_bag.zip")
+    dst = os.path.join(tmp_dir, "")
+    return shutil.copy(src, dst)
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def package(location, bag_fixture):
+    return models.Package.objects.create(
+        current_location=location,
+        current_path=os.path.join(location.full_path, bag_fixture),
```

Comment:
If I insert a `breakpoint()` before this return, the value of the `current_path` field looks like:

```
(Pdb) p os.path.join(location.full_path, bag_fixture)
'/tmp/pytest-of-user/pytest-0/test_get_fixity_check_report_s0/dir/working_bag.zip'
```

That's still an absolute path not relative. The value of `current_path` needs to be a relative path.

Reply:
I just tested this alternative approach:

* Create the location directory in the `location` fixture
* Copy the fixture file in the `package` fixture

See how `current_path` is set to a relative path when creating the `Package` model instance.

```diff
diff --git a/tests/locations/test_package.py b/tests/locations/test_package.py
index 7e235b5..0468e14 100644
--- a/tests/locations/test_package.py
+++ b/tests/locations/test_package.py
@@ -1396,6 +1396,9 @@ def location(space):
         relative_path="fs-aips",
         purpose="AS",
     )
+
+    pathlib.Path(aipstore.full_path).mkdir()
+
     return aipstore
 
 
@@ -1407,25 +1410,21 @@ def internal_location(space):
     )
 
 
-@pytest.fixture
-def bag_fixture(tmp_path):
-    tmp_dir = tmp_path / "dir"
-    tmp_dir.mkdir()
-    src = os.path.join(FIXTURES_DIR, "working_bag.zip")
-    dst = os.path.join(tmp_dir, "")
-    return shutil.copy(src, dst)
-
-
 @pytest.fixture
 @pytest.mark.django_db
-def package(location, bag_fixture):
-    return models.Package.objects.create(
+def package(location):
+    result = models.Package.objects.create(
         current_location=location,
-        current_path=os.path.join(location.full_path, bag_fixture),
+        current_path="working_bag.zip",
         package_type="AIP",
         status="Uploaded",
     )
 
+    src = os.path.join(FIXTURES_DIR, "working_bag.zip")
+    shutil.copy(src, result.full_path)
+
+    return result
+
 
 @pytest.mark.django_db
 @mock.patch(
```

Reply:
Ahhh...well understood. Thanks!